include/illixr/runge-kutta.hpp

```diff
@@ -0,0 +1,197 @@
+#pragma once
+#include "ProperQuaternion.hpp"
+
+#include <eigen3/Eigen/Dense>
+#include <iostream>
+
+namespace ILLIXR {
+
+const ProperQuaterniond dq_0(1., 0., 0., 0.);    /** Initial quaternion*/
+const Eigen::Vector3d   Gravity(0.0, 0.0, 9.81); /** Gravitational acceleration, at sea level, on Earth*/
+
+/**
+ * @brief Generates a skew-symmetric matrix from the given 3-element vector
+ *
+ * Based on equation 6 from http://mars.cs.umn.edu/tr/reports/Trawny05b.pdf
+ */
+inline Eigen::Matrix3d symmetric_skew(const Eigen::Vector3d& vec) {
+    Eigen::Matrix3d skew_q;
+    skew_q << 0, -vec[2], vec[1], vec[2], 0, -vec[0], -vec[1], vec[0], 0;
+    return skew_q;
+}
+
+/**
+ * @brief Generate the Omega matrix from the input 3-element vector
+ *
+ * Based on equation 48 of http://mars.cs.umn.edu/tr/reports/Trawny05b.pdf
+ */
+inline Eigen::Matrix4d makeOmega(const Eigen::Vector3d& w) {
+    Eigen::Matrix4d omega   = Eigen::Matrix4d::Zero();
+    omega.block<3, 3>(0, 0) = -symmetric_skew(w);
+    omega.block<1, 3>(3, 0) = -w.transpose();
+    omega.block<3, 1>(0, 3) = w;
+    return omega;
+}
+
+/**
+ * Calculate the change in orientation based on the input Quaternion
+ * @param k_n The input Quaternion
+ * @return The change in orientation
+ */
+inline ProperQuaterniond delta_q(const ProperQuaterniond& k_n) {
+    ProperQuaterniond dq(dq_0 + 0.5 * k_n);
+    dq.normalize();
+    return dq;
+}
+
+/**
+ * f(x) for the orientation quaternion
+ * @param av Vector representing the linear acceleration
```

Comment:
I'm not following this comment, it seems like only angular velocity gets passed into this function.